yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,566 @@
+    type enumeration {
+      enum "unknown" {
+        value 0;
+        configd:help "Unknown network type. Only valid as a state.";
+        description "Unknown network type. Only valid as a state.";
+      }
+      enum "broadcast" {
+        value 1;
+        configd:help "Broadcast circuit network-type.";
+        description "Broadcast circuit network-type.";
+      }
+      enum "point-to-point" {
+        value 2;
+        configd:help "Point-to-point circuit network-type."; 
+        description "Point-to-point circuit network-type.";
+      }
+      enum "loopback" {
+        value 3;
+        configd:help "Loopback circuit network-type. Only valid as a state.";
+        description "Loopback circuit network-type. Only valid as a state.";
+      }
+    }
+  }
+
+  typedef snpa {
+    type string {
+      length "0 .. 20";
+    }
+    configd:help "This type defines the Subnetwork Point of Attachment (SNPA) format.
+    The SNPA should be encoded according to the rules specified for the particular type
+    of subnetwork being used. As an example, for an ethernet subnetwork, the SNPA is 
+    encoded as a MAC address like '00aa.bbcc.ddee'.";
+
+    description "This type defines the Subnetwork Point of Attachment (SNPA) format.
+    The SNPA should be encoded according to the rules specified for the particular type
+    of subnetwork being used. As an example, for an ethernet subnetwork, the SNPA is
+    encoded as a MAC address like '00aa.bbcc.ddee'.";
+  }
+
+  typedef net-address {
+    type string {
+      pattern "[a-fA-F0-9]{2}(\\.[a-fA-F0-9]{4}){3,9}\\.[a-fA-F0-9]{2}";
+    }
+    configd:help "This type defines an OSI NET address using a pattern,
+    An example net-address is 49.0123.6452.1972.00";
+    description "This type defines an OSI NET address using a pattern,
+    An example net-address is 49.0123.6452.1972.00";
+  }
+
+  typedef metric-style-type {
+    type enumeration {
+      enum "narrow" {
+        value 0;
+        configd:help "This enum describes narrow metric style";
+        description "This enum describes narrow metric style";
+        reference "RFC1195";
+      }
+      enum "wide" {
+        value 1;
+        configd:help "This enum describes wide metric style";
+        description "This enum describes wide metric style";
+        reference "RFC5305";
+      }
+      enum "transition" {
+        value 2;
+        configd:help "This enum describes transition metric style";
+        description "This enum describes transition metric style";
+      }
+    }
+  }
+
+  typedef timer-val {
+    type uint16 {
+      range "0..60000";
+    }
+  }
+
+  grouping isis-password {
+    leaf clear {
+      configd:help "Clear-text authentication type.";
+      description "Clear-text authentication type.";
+      type string {
+        length "1..254";
+      }
+      must "not(../md5)" {
+        error-message "Can't configure both clear and md5 for password";
+      }
+    }
+    leaf md5 {
```

Comment:
Please also add `configd:secret "true";` to this leaf.

Reply:
Sure, will add it. 

Thanks,

Reply:
This is Resolved.